# HG changeset patch
# User Ben Abbott  <bpabbott@mac.com>
# Date 1297212403 18000
#      Tue Feb 08 19:46:43 2011 -0500
# Branch release-3-4-x
# Node ID 9b3e17b892b4ba793162c216241ab05215e3574c
# Parent  2c31574f5fe8282a7a2d874aa9e935709671d0d4
__go_draw_axes__.m: Properly set fontspec for legends.

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2011-02-08  Ben Abbott  <bpabbott@mac.com>
+
+	* plot/__go_draw_axes__.m: Properly set fontspec for legends.
+
 2011-02-05  David Bateman  <dbateman@free.fr>
 
 	* plot/legend.m: Allow the location and orientation to be set without
 	modifiying the legend keys.
 	* plot/legend.m: Add padding to legend positions. Fix for outerposition
 	for southeastoutside and southwestoutside.
 
 2011-02-06  Ben Abbott  <bpabbott@mac.com>
diff --git a/scripts/plot/__go_draw_axes__.m b/scripts/plot/__go_draw_axes__.m
--- a/scripts/plot/__go_draw_axes__.m
+++ b/scripts/plot/__go_draw_axes__.m
@@ -1467,17 +1467,17 @@ function __go_draw_axes__ (h, plot_strea
           pos = "";
           warning ("legend: 'Best' not yet implemented for location specifier.\n");
           ## Least conflict with data in plot.
           ## Least unused space outside plot.
         otherwise
           pos = "";
       endswitch
       if (__gnuplot_has_feature__ ("key_has_font_properties"))
-        [fontname, fontsize] = get_fontname_and_size (obj);
+        [fontname, fontsize] = get_fontname_and_size (hlgnd);
         fontspec = create_fontspec (fontname, fontsize, gnuplot_term);
       else
         fontspec = "";
       endif
       fprintf (plot_stream, "set key %s %s;\nset key %s %s %s %s;\n",
                inout, pos, box, reverse, horzvert, fontspec);
     else
       fputs (plot_stream, "unset key;\n");
